ingestion/src/metadata/ingestion/source/database/vertica/query_parser.py

```diff
@@ -0,0 +1,69 @@
+#  Copyright 2021 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""
+Vertica usage module
+"""
+from abc import ABC
+from typing import Iterable
+
+from metadata.generated.schema.entity.services.connections.database.verticaConnection import (
+    VerticaConnection,
+)
+from metadata.generated.schema.entity.services.connections.metadata.openMetadataConnection import (
+    OpenMetadataConnection,
+)
+from metadata.generated.schema.metadataIngestion.workflow import (
+    Source as WorkflowSource,
+)
+from metadata.generated.schema.type.tableQuery import TableQuery
+from metadata.ingestion.api.source import InvalidSourceException
+from metadata.ingestion.source.connections import get_connection
+from metadata.ingestion.source.database.query_parser_source import QueryParserSource
+from metadata.ingestion.source.database.vertica.queries import VERTICA_LIST_DATABASES
+from metadata.utils.logger import ingestion_logger
+
+logger = ingestion_logger()
+
+
+class VerticaQueryParserSource(QueryParserSource, ABC):
+    """
+    Vertica lineage parser source.
+
+    Vertica V_MONITOR schema changes from database to database.
+    To allow the lineage to happen for all the ingested databases
+    we'll need to iterate over them.
+    """
+
+    filters: str
+
+    @classmethod
+    def create(cls, config_dict, metadata_config: OpenMetadataConnection):
+        """Create class instance"""
+        config: WorkflowSource = WorkflowSource.parse_obj(config_dict)
+        connection: VerticaConnection = config.serviceConnection.__root__.config
+        if not isinstance(connection, VerticaConnection):
+            raise InvalidSourceException(
+                f"Expected VerticaConnection, but got {connection}"
+            )
+        return cls(config, metadata_config)
+
+    def get_table_query(self) -> Iterable[TableQuery]:
+        database = self.config.serviceConnection.__root__.config.database
+        if database:
+            yield from super().get_table_query()
```

Comment:
I believe this will call `get_table_query` from `QueryParserSource` which does not implement the method

Reply:
So here we actually have some nasty inheritance working that we might want to simplify in the future because tbh, now that you said it, it has me a bit lost.

Our classes at play are:
- `VerticaLineageSource(VerticaQueryParserSource, LineageSource)`
- `VerticaQueryParserSource(QueryParserSource, ABC)` -> implements `get_table_query` and calls `super().get_table_query()`
- `LineageSource(QueryParserSource, ABC)` -> implements `get_table_query`

So when we run `VerticaLineageSource` it runs the `VerticaQueryParserSource.get_table_query` that then runs `LineageSource.get_table_query`

This works because of the inheritance order of the classes set in `VerticaLineageSource(VerticaQueryParserSource, LineageSource)`. Changing the order does not follow the right calls.

I am sure we can find a less dark way to control the flow here. IMO we can open another issue to track it separated, as this is implemented in other lineage/usage sources as well. Thanks for pointing it out

Reply:
I see. That makes sense. it follows the mro of `VerticaLineageSource` so it will call the `get_table_query` of `LineageSource `.

Reply:
we're just reusing one implementation of the query parsing for lineage and usage sources. We can come up with a better way. opened https://github.com/open-metadata/OpenMetadata/issues/9936 to track this.

IMO for now we should be able to merge this one and then update in bulk